roles/rsyslog/tasks/main_core.yml

```diff
@@ -292,6 +292,37 @@
       when: __rsyslog_enabled | bool
       notify: restart rsyslogd
 
+    - name: Manage firewall for specified ports
+      include_role:
+        name: fedora.linux_system_roles.firewall
+      vars:
+        firewall: "{{ logging_firewall }}"
+      when: logging_firewall | length > 0
+
+    - name: "Prepare _logging_selinux_ports to drop ports defined
+            in the selinux policy"
+      set_fact:
+        _logging_selinux_ports: []
+
+    - name: Drop ports defined in the selinux policy
```

Comment:
What happens when you pass in ports that are already defined in the selinux policy?

Reply:
> What happens when you pass in ports that are already defined in the selinux policy?

Add is ok. Just silently skipped ("changed": false). But if I try to delete them with these parameters:
```
logging_selinux_ports:
   - ports: 1514
     proto: tcp
     setype: syslogd_port_t
     state: absent
   - ports: 20514
     proto: tcp
     setype: syslogd_port_t
     state: absent
```
the selinux role (in the collection) fails like this:
```
TASK [fedora.linux_system_roles.selinux : Set an SELinux label on a port] ******
<<snip>>
failed: [/home/nhosoi/.cache/linux-system-roles/rhel-9.qcow2] (item={'ports': 20514, 'proto': 'tcp', 'setype': 'syslogd_port_t', 'state': 'absent'}) => {
    "ansible_loop_var": "item",
    "changed": false,
    "item": {
        "ports": 20514,
        "proto": "tcp",
        "setype": "syslogd_port_t",
        "state": "absent"
    }
}

MSG:
ValueError: Port tcp/20514 is defined in policy, cannot be deleted

RUNNING HANDLER [/home/nhosoi/linux-system-roles/logging/tests/roles/linux-system-roles.logging/roles/rsyslog : restart rsyslogd] ***
task path: /home/nhosoi/linux-system-roles/logging/roles/rsyslog/handlers/main.yml:2
Friday 19 August 2022  15:57:30 -0700 (0:00:02.422)       0:01:53.744 ********* 

PLAY RECAP *********************************************************************
/home/nhosoi/.cache/linux-system-roles/rhel-9.qcow2 : ok=126  changed=20   unreachable=0    failed=1    skipped=114  rescued=0    ignored=0   
```

Reply:
> > What happens when you pass in ports that are already defined in the selinux policy?
> 
> Add is ok. Just silently skipped ("changed": false). But if I try to delete them with these parameters:
> 
> ```
> logging_selinux_ports:
>    - ports: 1514
>      proto: tcp
>      setype: syslogd_port_t
>      state: absent
>    - ports: 20514
>      proto: tcp
>      setype: syslogd_port_t
>      state: absent
> ```
> 
> the selinux role (in the collection) fails like this:
> 
> ```
> TASK [fedora.linux_system_roles.selinux : Set an SELinux label on a port] ******
> <<snip>>
> failed: [/home/nhosoi/.cache/linux-system-roles/rhel-9.qcow2] (item={'ports': 20514, 'proto': 'tcp', 'setype': 'syslogd_port_t', 'state': 'absent'}) => {
>     "ansible_loop_var": "item",
>     "changed": false,
>     "item": {
>         "ports": 20514,
>         "proto": "tcp",
>         "setype": "syslogd_port_t",
>         "state": "absent"
>     }
> }
> 
> MSG:
> ValueError: Port tcp/20514 is defined in policy, cannot be deleted
> 
> RUNNING HANDLER [/home/nhosoi/linux-system-roles/logging/tests/roles/linux-system-roles.logging/roles/rsyslog : restart rsyslogd] ***
> task path: /home/nhosoi/linux-system-roles/logging/roles/rsyslog/handlers/main.yml:2
> Friday 19 August 2022  15:57:30 -0700 (0:00:02.422)       0:01:53.744 ********* 
> 
> PLAY RECAP *********************************************************************
> /home/nhosoi/.cache/linux-system-roles/rhel-9.qcow2 : ok=126  changed=20   unreachable=0    failed=1    skipped=114  rescued=0    ignored=0   
> ```

I see.  This is going to be tricky.  I can think of a couple of ways to do this:

* use a `register` and a `failed_when`
```
  register: __logging_seport
  failed_when: not __logging_seport.msg is search('is defined in policy, cannot be deleted')
```
I'm not sure which field in the register variable holds the message.  If register/failed_when will simply not work, then try a block/rescue

* use `semanage port -C --list` - parse the list into a form you can use to compare against the values in logging_selinux_ports, then only apply the ones in that list

Reply:
Thank you, @richm. Let me try your ideas.

BTW, I guess it's too early for me to set `[citest]`. :) We need our citool pr merged first to make the ci tests in this pr pass...

Reply:
> Thank you, @richm. Let me try your ideas.

Thinking about this some more
* Using register/failed_when probably won't work correctly because we're using it with the `include_role`, and there may be multiple ports we want to try to disable - would have to be in a loop to try each port separately - same with block/rescue
* Using `semanage port -C --list` means having to ensure that the `semanage` command is installed - not sure if the `selinux` role does that
* The easiest way would be if the `selinux` role had an option to allow you to ignore errors when trying to remove built-in policy - that way you could just pass in all of them and ignore all of the errors

> 
> BTW, I guess it's too early for me to set `[citest]`. :) We need our citool pr merged first to make the ci tests in this pr pass...

There are several roles now that need collection-requirements.yml